Improving root level readme (#1470)
connext/nxtp: README.md

    <br />
    <a href="https://immunefi.com/bounty/connext/">Bug Bounty Program</a>
    <br />
    <br />
    <a href="https://github.com/connext/nxtp/issues">Request Feature</a>
  </p>
</div>

<!-- TABLE OF CONTENTS -->
<details>
  <summary>Table of Contents</summary>
  <ol>
    <li>
      <a href="#about-nxtp">About NXTP </a>
       <ul>
        <li><a href="#built-with">Architecture</a></li>
      </ul>
      <ul>
        <li><a href="#built-with">Architecture</a></li>
      </ul>
    </li>
    <li>
      <a href="#getting-started">Getting Started</a>
      <ul>
        <li><a href="#prerequisites">Prerequisites</a></li>
        <li><a href="#installation">Installation</a></li>
      </ul>
    </li>
    <li><a href="#usage">Usage</a></li>
    <li><a href="#roadmap">Roadmap</a></li>
    <li><a href="#contributing">Contributing</a></li>
    <li><a href="#license">License</a></li>
    <li><a href="#contact">Contact</a></li>
    <li><a href="#acknowledgments">Acknowledgments</a></li>
  </ol>
</details>

<!-- ABOUT THE PROJECT -->

## About Connext

Connext is a bridging protocol which allows for safe transfer of tokens and data between EVM compatible blockchains.

> It's architecture can be seen as a layered system, as follows:

Reviewer: ```suggestion
The Connext architecture can be seen as a layered system, as follows:
```
"It's" is improper grammar.

| `Liquidity Layer`       | `NXTP`                            |
| `Gateway/Routing Layer` | `Interchain Gateway Protocol`     |
| `Messaging Layer`       | `Nomad`                           |
| `Transport Layer`       | `Connext Routers`                 |

## Core Flow of a transaction through Connext

A transaction flowing through Connext will now have the following lifecycle:

**A User will:**

Initiate the transaction by calling a `xcall` function on our contracts, passing in funds, gas details, arbitrary data, and a target address object (includes chain info). Note that `xcall` is meant to mimic solidity's lower level `call` as best as possible.

**Our contracts will:**

-If needed, swap the passed in token to the Nomad version of the same asset.
Call the Nomad contracts with a hash of the tx details to initiate the 30-60m message across chains.

-Emit an event with the tx details.
Routers observing the origin chain with funds on the destination chain will:
Simulate the transaction (if this fails, the assumption is that this is a more "expressive" crosschain message that requires permissioning and so must go through the slow Nomad process only).

> -Prepare a signed transaction object using funds on the receiving chain.

Reviewer: ```suggestion
**Routers (Active Liquidity Providers) will:**
-Prepare a signed transaction object using funds on the receiving chain.
```
This is done by routers technically, not contracts.

Post this object (a "bid") to the auctioneer.

> Note: if the router does not have enough funds for the transfer, they may also provide only part of the transfer's value.

Reviewer: ```suggestion
Note: if the router does not have enough funds for the transfer, they may also provide only part of the transfer's value, which gets split across multiple routers in the network.
```

> -The auctioneer will be observing all of the underlying chains. Every X blocks, the auctioneer will collect bids for transactions. The auctioneer will be responsible for selecting the correct router (or routers!) for a given tx (can be random). The auctioneer will post batches of these bids to a relayer network to submit them to chain.

Reviewer: ```suggestion
-The sequencer collects bids from routers and allows routers 30 seconds per transfer to send bids. After this time, sequencer will select a bid randomly (selection process TBD) and submit the payload which contains the router's signature to the relayer network to be submitted to chain.
```
Modified sequencer algorithm to be more accurate.

--> At this point, the user's tx has already been completed!

Later, when the Nomad message arrives, a heavily batched tx can be submitted to take all pending hashes received over Nomad and look up whether they have corresponding router addresses in the hash -> router address mapping. If they do, then Nomad assets are minted and given to the router.

Note: if the router gives the incorrect amount of funds to a user or if they execute the wrong calldata, then the router's param hash will not match the hash coming over Nomad and the router will not get reimbursed. This is the core security mechanism that ensures that routers behave correctly.

Note 2: Routers will take a 30-60 minute lockup on their funds when relaying transactions. While this theoretically reduces capital efficiency compared to the existing system, in practice the lack of need to rebalance will mean that routers have more capital available more often regardless.

## About NXTP

**Nxtp** is a liquidity layer and a developer interface on top of the [nomad](nomad-url) optimisitic briding protocol.

<p align="right">(<a href="#top">back to top</a>)</p>

### Structure of this repo

> This repo is set up as a yarn workspace monorepo

Reviewer: ```suggestion
This repo is set up as a Yarn Workspace monorepo: https://yarnpkg.com/features/workspaces
```
